runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -59,6 +60,61 @@ class MemoryManager final {
     bool contains(void *ptr) { return _impl.contains(ptr); }
 };
 
+class SharedLibraryManager final {
+  private:
+    void *_handler{NULL};
+
+  public:
+    SharedLibraryManager() = delete;
+    SharedLibraryManager(std::string filename)
```

Comment:
Making this constructor as 'explicit', avoid implicit conversions and copy initializations. 
```suggestion
    explicit SharedLibraryManager(std::string filename)
```
Check the ref here: https://en.cppreference.com/w/cpp/language/explicit